File: Source/MKL/examples/vmlc/source/callback.c
#include <stdio.h>
#include <math.h>
#include "func_interv.h"

#include "mkl_vml.h"

int __stdcall StubCallBack(DefVmlErrorContext* pdefVmlErrorContext)
{
  return 0;
}

int main()
{
  VMLErrorCallBack errcb;

  printf("vmlSetErrorCallBack/vmlGetErrorCallBack/vmlClearErrorCallBack example programm\n\n");

#if defined(MKL_VML_MIXED)
  errcb=vmlGetErrorCallBack();
#elif defined(MKL_VML_LOWER)
  errcb=vmlgeterrorcallback();
#elif defined(MKL_VML_UPPER)
  errcb=VMLGETERRORCALLBACK();
#else
# error You must define MKL_VML_CDECL for CDECL call convention or MKL_VML_STDCALL for STDCALL call convention.
#endif
  printf("Initial address of CallBack function: 0x%X%X\n",errcb);

  errcb=StubCallBack;
#if defined(MKL_VML_MIXED)
  vmlSetErrorCallBack(errcb);
#elif defined(MKL_VML_LOWER)
  vmlseterrorcallback(errcb);
#elif defined(MKL_VML_UPPER)
  VMLSETERRORCALLBACK(errcb);
#else
# error You must define MKL_VML_CDECL for CDECL call convention or MKL_VML_STDCALL for STDCALL call convention.
#endif
  errcb=vmlGetErrorCallBack();
  printf("Address of CallBack function after using vmlSetErrorCallBack: 0x%X%X\n",errcb);

#if defined(MKL_VML_MIXED)
  vmlClearErrorCallBack();
#elif defined(MKL_VML_LOWER)
  vmlclearerrorcallback();
#elif defined(MKL_VML_UPPER)
  VMLCLEARERRORCALLBACK();
#else
# error You must define MKL_VML_CDECL for CDECL call convention or MKL_VML_STDCALL for STDCALL call convention.
#endif
  errcb=vmlGetErrorCallBack();
  printf("Address of CallBack function after using vmlClearErrorCallBack: 0x%X%X\n",errcb);
}

File: Source/MKL/examples/vmlf/source/callback.f
	programm MKL_VML_TEST

	include "mkl_vml.fi"
	include "func_interv.fi"

	integer*4 VMLGETERRORCALLBACK
	integer*4 errcb

	print *,"vmlSetErrorCallBack/vmlGetErrorCallBack/vmlClearErrorCallBack example programm"
	print *,""
	print *,""

	errcb=VMLGETERRORCALLBACK()
	print 10,"Initial address of CallBack function: ",errcb,"h"

	call VMLSETERRORCALLBACK(StubCallBack)
	errcb=VMLGETERRORCALLBACK()
	print 10,"Address of CallBack function after using vmlSetErrorCallBack: ",errcb,"h"

	call VMLCLEARERRORCALLBACK()
	errcb=VMLGETERRORCALLBACK()
	print 10,"Address of CallBack function after using vmlClearErrorCallBack: ",errcb,"h"

10	format (A,Z,A)

	end

	
	integer function StubCallBack(p)

	StubCallBack=0

	end
